tasks.py

```diff
@@ -60,5 +72,3 @@ def run(self):
         if not self.merge_request(number):
             print('An error occurred when trying to merge the pull request')
```

Comment:
This could be a bit misleading...it's possible the merge succeeded while the delete failed. Maybe make a separate function for the `delete_request` and handle that as a separate step?

Reply:
Sounds good. I refactored it to better match the existing patterns.

Reply:
Nice refactoring. That's exactly what I was thinking.

In testing this, the `delete_branch` function is returning a 404 for me. Do you think it is a permissions error?

Reply:
I'd expect a 403 for a permissions error. It's actually a typo in the url. It should be `git` rather than `get`

Reply:
Nice - it works now but needs a small tweak. Looks like the delete is returning `204` rather than `200`. https://docs.github.com/en/free-pro-team@latest/rest/reference/git#delete-a-reference